tools/maint/win32_launcher/launcher.c

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2025 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ *
+ * Small win32 application that is used to launcher emscripten via python.exe.
+ * On non-windows platforms this is done via the run_pyton.sh shell script.
+ *
+ * The binary will look for a python script that matches its own name and run
+ * that using python.exe.
+ */
+
+// Define _WIN32_WINNT to Windows 7 for max portability
+#define _WIN32_WINNT 0x0601
+
+#include <windows.h>
+#include <shellapi.h>
+#include <shlwapi.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#pragma comment(lib, "shlwapi.lib")
+#pragma comment(lib, "shell32.lib")
+
+wchar_t* get_python_executable() {
+  wchar_t* python_exe_w = _wgetenv(L"EMSDK_PYTHON");
+  if (!python_exe_w) {
+    return L"python";
+  }
+  return python_exe_w;
+}
+
+wchar_t* get_long_path(const wchar_t* path) {
+  DWORD path_len = GetFullPathNameW(path, 0, NULL, NULL);
+  if (path_len == 0)
+    abort();
+
+  wchar_t* full_path = malloc(sizeof(wchar_t) * (path_len + 4));
+  if (!full_path)
+    abort();
+
+  wcscpy_s(full_path, path_len + 4, L"\\\\?\\");
+  if (GetFullPathNameW(path, path_len, full_path + 4, NULL) == 0)
+    abort();
+  return full_path;
+}
+
+// Get the name of the currently running executable (module)
+wchar_t* get_module_path() {
```

Comment:
Is this necessary vs just `argv[0]`?

Reply:
I think on windows there are lot of subtleties here, and this is the most accurate/offiical/windowsy way to do it.

Using argc/argv comes from the unix world and IIRC is kind of layered on top in the windows world.  

Hoestly though I'm not an expert and I had AI help me write this, with an emphasis on handling as many strange corner cases and windows-isms as possible.

Reply:
If you google "GetModuleFileNameW vs just using argv0" there are apparently a bunch of good reasons for this.